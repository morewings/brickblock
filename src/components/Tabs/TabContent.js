import React from 'react';
import PropTypes from 'prop-types';
import {TabContent as Content, TabPane} from 'reactstrap';
import Table from 'components/Table/Table';
import Pages from 'components/Pages/Pages.container';

const TabContent = props => (
  <Content activeTab={props.activeTab}>
    <TabPane tabId="tab1">
      <Table />
      <Pages dataSource="preIco" />
    </TabPane>
    <TabPane tabId="tab2">
        Tab 2
    </TabPane>
  </Content>
);

TabContent.propTypes = {
  activeTab: PropTypes.string.isRequired,
};

export default TabContent;
