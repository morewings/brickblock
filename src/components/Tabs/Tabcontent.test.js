import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import TabContent from 'components/Tabs/TabContent';

describe('components > Tabs > TabContent', () => {
  it('renders without crashing with `props.activeTab === \'tab1\'`', () => {
    const defaultProps = {
      activeTab: 'tab1',
    };
    const wrapper = shallow(<TabContent {...defaultProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders without crashing with `props.activeTab === \'tab2\'`', () => {
    const defaultProps = {
      activeTab: 'tab2',
    };
    const wrapper = shallow(<TabContent {...defaultProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
