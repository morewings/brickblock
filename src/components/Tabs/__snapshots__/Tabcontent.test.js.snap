// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Tabs > TabContent renders without crashing with \`props.activeTab === 'tab1'\` 1`] = `
<TabContent
  activeTab="tab1"
  tag="div"
>
  <TabPane
    tabId="tab1"
    tag="div"
  >
    <Table />
    <Connect(Pages)
      dataSource="preIco"
    />
  </TabPane>
  <TabPane
    tabId="tab2"
    tag="div"
  >
    Tab 2
  </TabPane>
</TabContent>
`;

exports[`components > Tabs > TabContent renders without crashing with \`props.activeTab === 'tab2'\` 1`] = `
<TabContent
  activeTab="tab2"
  tag="div"
>
  <TabPane
    tabId="tab1"
    tag="div"
  >
    <Table />
    <Connect(Pages)
      dataSource="preIco"
    />
  </TabPane>
  <TabPane
    tabId="tab2"
    tag="div"
  >
    Tab 2
  </TabPane>
</TabContent>
`;
